packages/utils-query-operation/lib/MaterializeBindings.ts

```diff
@@ -106,9 +106,29 @@ export function materializeOperation(
           };
         }
       }
+      let recursionResult: Algebra.Operation = materializeOperation(
+        op.input,
+        bindings,
+        algebraFactory,
+        bindingsFactory,
+        options,
+      );
+
+      // Join op.input with a values clause containing the variables that are in InitialBindings and
+      // are used in the current extend operation.
+      const inScopeVariables = [ op.variable ];
+      if (op.expression.expressionType === Algebra.expressionTypes.TERM && op.expression.term.termType === 'Variable') {
+        inScopeVariables.push(op.expression.term);
+      }
+      const values: Algebra.Operation[] =
+        createValuesFromBindings(factory, <Bindings> options.originalBindings, inScopeVariables);
```

Comment:
The cast to `Bindings` may break if `options.originalBindings` is undefined.

Reply:
I thought it would never be undefined because options.originalBindings is assigned like this
`originalBindings: 'originalBindings' in options ? options.originalBindings : bindings`.
Should I just assert it by adding a null check or what is the best way?

Reply:
Ah yes, you're right, it should be fine!